icechunk/src/format/manifest.rs

```diff
@@ -215,15 +221,14 @@ impl Iterator for PayloadIterator {
                 if let Some((k @ (_, coord), payload)) = self
                     .manifest
                     .chunks
-                    .range((Bound::Excluded(last_key), Bound::Unbounded))
+                    .range((
+                        Bound::Excluded(last_key.clone()),
+                        Bound::Included((self.for_node.clone(), upper_bound)),
+                    ))
                     .next()
                 {
-                    if k.0 == self.for_node {
-                        self.last_key = Some(k.clone());
-                        Some((coord.clone(), payload.clone()))
-                    } else {
-                        None
-                    }
+                    self.last_key = Some(k.clone());
+                    Some((coord.clone(), payload.clone()))
```

Comment:
With the `take_while` approach, we would return `ChunkInfo` here and unpack it up in `verified_node_chunk_iterator`. Worth it?